src/api/AppApi.test.js

```diff
@@ -176,3 +197,394 @@ function testInstallApp() {
+    testApiShouldNotThrowOnInvalidParameters(fnToTest, validParams, invalidParams);
+    testApiShouldRejectOnInvalidParameters(fnToTest, validParams, invalidParams);
+    testApiShouldSendCorrectGetRequest(fnToTest, validParams, axiosParams);
+  });
+}
+
+function testRemoveAppTypeFromApp() {
+
+  describe('removeAppTypeFromApp()', () => {
+
+    const fnToTest = AppApi.removeAppTypeFromApp;
+
+    const validParams = [MOCK_APP_DM.id, MOCK_APP_TYPE_DM.id];
+    const invalidParams = [INVALID_PARAMS_SS, INVALID_PARAMS_SS];
+    const axiosParams = [`/${UPDATE_PATH}/${MOCK_APP_DM.id}/${MOCK_APP_TYPE_DM.id}`];
+
+    testApiShouldReturnPromise(fnToTest, validParams);
+    testApiShouldUseCorrectAxiosInstance(fnToTest, validParams, APP_API);
+    testApiShouldNotThrowOnInvalidParameters(fnToTest, validParams, invalidParams);
+    testApiShouldRejectOnInvalidParameters(fnToTest, validParams, invalidParams);
+    testApiShouldSendCorrectDeleteRequest(fnToTest, validParams, axiosParams);
+  });
+}
+
+function testUpdateAppEntitySetConfig() {
+
+  describe('updateAppEntitySetConfig()', () => {
+
+    const fnToTest = AppApi.updateAppEntitySetConfig;
+
+    const validParams = [MOCK_ORGANIZATION_DM.id, MOCK_APP_DM.id, MOCK_APP_TYPE_DM.id, MOCK_ENTITY_SET_DM.id];
+    const invalidParams = [INVALID_PARAMS_SS, INVALID_PARAMS_SS, INVALID_PARAMS_SS, INVALID_PARAMS_SS];
+    const axiosParams = [
+      `/${UPDATE_PATH}/${MOCK_ORGANIZATION_DM.id}/${MOCK_APP_DM.id}/${MOCK_APP_TYPE_DM.id}/${MOCK_ENTITY_SET_DM.id}`];
+
+    testApiShouldReturnPromise(fnToTest, validParams);
+    testApiShouldUseCorrectAxiosInstance(fnToTest, validParams, APP_API);
+    testApiShouldNotThrowOnInvalidParameters(fnToTest, validParams, invalidParams);
+    testApiShouldRejectOnInvalidParameters(fnToTest, validParams, invalidParams);
+    testApiShouldSendCorrectGetRequest(fnToTest, validParams, axiosParams);
+  });
+}
+
+function testUpdateAppConfigPermissions() {
+
+  describe('updateAppConfigPermissions()', () => {
+
+    const fnToTest = AppApi.updateAppConfigPermissions;
+    const permissions = [PermissionTypes.READ, PermissionTypes.WRITE];
+
+    const validParams = [MOCK_ORGANIZATION_DM.id, MOCK_APP_DM.id, MOCK_APP_TYPE_DM.id, permissions];
+    const invalidParams = [
+      INVALID_PARAMS_SS, INVALID_PARAMS_SS, INVALID_PARAMS_SS, [INVALID_PARAMS_SS, INVALID_PARAMS_SS]];
+    const axiosParams = [
+      `/${UPDATE_PATH}/${MOCK_ORGANIZATION_DM.id}/${MOCK_APP_DM.id}/${MOCK_APP_TYPE_DM.id}`,
+      permissions];
+
+    testApiShouldReturnPromise(fnToTest, validParams);
+    testApiShouldUseCorrectAxiosInstance(fnToTest, validParams, APP_API);
+    testApiShouldNotThrowOnInvalidParameters(fnToTest, validParams, invalidParams);
+    testApiShouldRejectOnInvalidParameters(fnToTest, validParams, invalidParams);
+    testApiShouldSendCorrectPostRequest(fnToTest, validParams, axiosParams);
+  });
+}
+
+function testUpdateAppMetadata() {
```

Comment:
This is intense :) `AppApi` is not the only place where this happens, so generally speaking, we need a better way of testing parameters like this. Unfortunately, I haven't put in the time to figure it out.

I recommend simplifying this down to a single case (not 7), like the other functions you've added. We can come back to this later when we have a few spare cycles to think.